### ISSUE TITLE
Cannot add migration with EF

### ISSUE THREAD

**Issue by TETYYS** [Barsonax/Singularity#125]
If Singularity is used as a dependency container for the project using `UseServiceProviderFactory`, `dotnet ef migrations add` throws an error:

```
Singularity.Exceptions.NoConstructorException: Type Microsoft.EntityFrameworkCore.DbContextOptions did not contain any public constructor.
   at Singularity.DefaultConstructorResolver.StaticSelectConstructor(Type type)
   at Singularity.ConstructorResolverCache.<StaticSelectConstructor>b__5_0(Type t)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Singularity.ConstructorResolverCache.StaticSelectConstructor(Type type)
   at Singularity.ConstructorResolverExtensions.ResolveConstructorExpression(IConstructorResolver resolver, Type type)
   at Singularity.Graph.Resolvers.ConcreteServiceBindingGenerator.Resolve(IResolverPipeline graph, Type type)+MoveNext()
   at System.Collections.Generic.LargeArrayBuilder`1.AddRange(IEnumerable`1 items)
   at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Singularity.Graph.Resolvers.ResolverPipeline.TryGetDependency(Type type)
   at Singularity.Graph.Resolvers.ResolverPipeline.TryResolveAll(Type type)+MoveNext()
   at System.Linq.Enumerable.SelectEnumerableIterator`2.ToArray()
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Singularity.Graph.Resolvers.CollectionServiceBindingGenerator.Resolve[TElement](IResolverPipeline graph, Type type)+MoveNext()
   at Singularity.Graph.Resolvers.CollectionServiceBindingGenerator.Resolve(IResolverPipeline graph, Type type)+MoveNext()
   at System.Collections.Generic.LargeArrayBuilder`1.AddRange(IEnumerable`1 items)
   at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Singularity.Graph.Resolvers.ResolverPipeline.TryGetDependency(Type type)
   at Singularity.Graph.Resolvers.ResolverPipeline.GetDependency(Type type)
   at Singularity.Graph.Resolvers.ResolverPipeline.Resolve(Type type)
   at Singularity.Scoped.System.IServiceProvider.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetServices[T](IServiceProvider provider)
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.FindContextTypes()
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.FindContextType(String name)
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.CreateContext(String contextType)
   at Microsoft.EntityFrameworkCore.Design.Internal.MigrationsOperations.AddMigration(String name, String outputDir, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigrationImpl(String name, String outputDir, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigration.<>c__DisplayClass0_0.<.ctor>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.<>c__DisplayClass3_0`1.<Execute>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
```

Workaround: remove call to `UseServiceProviderFactory` just for adding the migration

**Comment by Barsonax** [comment]
Seems the ConcreteServiceBindingGenerator is missing a check for abstract classes. Adding this check will solve the error but still need to test more to ensure it is working as it should.

**Comment by TETYYS** [comment]
Works great 👍

**Comment by Barsonax** [comment]
Fixed in 0.17.0
